blockchains/erc20/erc20_worker.js

```diff
@@ -3,24 +3,40 @@ const Web3 = require('web3');
 const constants = require('./lib/constants');
 const { logger } = require('../../lib/logger');
 const { extendEventsWithPrimaryKey } = require('./lib/extend_events_key');
-
-let contractEditor = null;
-if (constants.CONTRACT_MODE !== 'vanilla') { contractEditor = require('./lib/contract_overwrite').contractEditor; }
-
-const { stableSort } = require('./lib/util');
+const { ContractOverwrite, changeContractAddresses, extractChangedContractAddresses } = require('./lib/contract_overwrite');
+const { stableSort, readJsonFile } = require('./lib/util');
 const BaseWorker = require('../../lib/worker_base');
 const { nextIntervalCalculator } = require('../eth/lib/next_interval_calculator');
 const { TimestampsCache } = require('./lib/timestamps_cache');
 const { getPastEvents } = require('./lib/fetch_events');
 const { initBlocksList } = require('../../lib/fetch_blocks_list');
 
+/**
+ * A simple non cryptographic hash function similar to Java's 'hashCode'
+ * @param input A string input
```

Comment:
Should be something like
```js
* @param {string} input
* @returns {number} A 32 bit non-negative integer
```
also, maybe something more specific than "string input"? Could it be a random string or it has limitations?